macros/sql/get_column_values.sql

```diff
@@ -11,19 +10,45 @@ Returns:
     A list of distinct values for the specified columns
 #}
 
-{% macro get_column_values(table, column, max_records=none) -%}
+{% macro get_columns(table, column, max_records=none, fail=True, default='blank') -%}
+
+{#-- Prevent querying of db in parsing mode. This works because this macro does not create any new refs. #}
+    {%- if not execute -%}
+        {{ return('') }}
+    {% endif %}
+{#--  #}
+
+    {%- set table_value = adapter.get_relation(database=table.database,
+                                          schema=table.schema,
+                                         identifier=table.identifier) -%}
+
```

Comment:
Let's move the `default`-related logic out of the statement.  Can you do something like:
```
{% if target_relation is none and default is none %}
  -- raise an exception here
{% elif target_relation is none and default is not none %}
  -- return here to short circuit evaluation of the macro
{% endif %}
```

Both `exceptions.raise_compiler_error` and `return()` will halt execution of this macro. While `exceptions.raise_compiler_error` will bubble up an error into the dbt-core code, `return()` will just yield execution back to the calling jinja code. You can do `{{ return(default) }}` to avoid running a `select` statement against the database here too!

Rather than querying the database, we can use the `return()` function to short-circuit evaluation of this macro. This has two benefits:
1. It avoids running an unnecessary query
2. It preserves the type supplied by the caller

So, If you execute:
```
{% set values = dbt_utils.get_column_values(ref('other_model'), 'type', default=[]) %}
```
The `values` variable will be a list object if `ref('other_model')` does not exist. The code you have here would execute `select '[]' as value`, which would return a string like `"[]"`, which would probably be harder to operate on.

Reply:
I've implemented this feedback! Thanks so much for it. I hadn't considered preserving type, but that makes so much sense. Just as an FYI, that does introduce weirdness about how to supply the default value _into_ the macro, so I've done that in the if/else statement surrounding value list